third_party/libwebrtc/webrtc/modules/congestion_controller/nada_owd_bwe.cc

```diff
@@ -216,334 +167,75 @@ DelayBasedBwe::Result NadaOwdBwe::IncomingPacketFeedbackVector(
-  while (!dmin_history_.empty() &&
-         dtmp < dmin_history_.back().second) {
-    dmin_history_.pop_back();
-  }
-
-  dmin_history_.push_back(std::make_pair(now_ms, dtmp));
-}
-
-
-// maintain history of recent pkt delay (dfwd) records
-void NadaOwdBwe::UpdateDelHistory(int64_t now_ms) {
-
-  // Remove expired data points from history.
-  while (!max_del_history_.empty() &&
-         now_ms - max_del_history_.front().first > kNadaBweParamLogwinMs) {
-    max_del_history_.pop_front();
-  }
-
-  // Typical sliding-window algorithm for logging maximum values:
-  // Pop values lower than current delay value before pushing the current delay value.
-  while (!max_del_history_.empty() &&
-         nada_d_fwd_ >= max_del_history_.back().second) {
-    max_del_history_.pop_back();
-  }
-
-  max_del_history_.push_back(std::make_pair(now_ms, nada_d_fwd_));
-}
-
-void NadaOwdBwe::UpdatePlrHistory(int64_t now_ms) {
-
-  // Remove expired data points from history.
-  while (!max_plr_history_.empty() &&
-         now_ms - max_plr_history_.front().first > kNadaBweParamLogwinMs) {
-
-    max_plr_history_.pop_front();
-  }
-
-  // Typical sliding-window algorithm for logging maximum values:
-  // Pop values lower than current PLR before pushing the current PLR
-  while (!max_plr_history_.empty() &&
-         nada_plr_ >= max_plr_history_.back().second) {
-    max_plr_history_.pop_back();
-  }
-
-  max_plr_history_.push_back(std::make_pair(now_ms, nada_plr_));
-}
-
 // We ignore updates to the RTT as one way delay (OWD) information is
 // more accurate for this version of the algorithm
 void NadaOwdBwe::OnRttUpdate(int64_t avg_rtt_ms, int64_t max_rtt_ms) {
+
+  printf("[DEBUG]: inside OnRttUpdate: rtt_avg = %8lld ms, rtt_max = %8lld ms\n", 
+          avg_rtt_ms, max_rtt_ms); 
+
+  RTC_LOG(LS_INFO) << "[DEBUG]  inside OnRttUpdate: "
+                   << "  rtt_avg =" << avg_rtt_ms 
+                   << " ms , rtt_max = " << max_rtt_ms
+                   << " ms" << std::endl; 
 }
 
 // This method can be implemented if need be, but current version of the
 // algorithm works well without it
 void NadaOwdBwe::SetStartBitrate(int start_bitrate_bps) {
+
+  printf("[DEBUG]: inside SetStartBitrate: start_bitrate = %d Kbps\n", 
+         start_bitrate_bps/1000); 
+
+  RTC_LOG(LS_INFO) << " [DEBUG]  inside SetStartBitrate: start_bitrate ="
+                   << start_bitrate_bps/1000 << " Kbps" << std::endl; 
 }
 
 // This method can be implemented if need be, but current version of the
 // algorithm works well without it
 void NadaOwdBwe::SetMinBitrate(int min_bitrate_bps) {
+
+  printf("[DEBUG]: inside SetMinBitrate: incoming min_bitrate = %d Kbps\n", 
+         min_bitrate_bps/1000); 
+
+  RTC_LOG(LS_INFO) << " [DEBUG] inside SetMinBitrate: incoming min_bitrate ="
+                   << min_bitrate_bps/1000 << " Kbps" << std::endl; 
 }
 
+// Note by Xiaoqing: it seems that this is uncalled so perhaps we can remove? 
```

Comment:
Good idea

Reply:
testing notification...

Reply:
Just in case... copy-pasting
An idea to check if it's used:
Add RTC_CHECK_LE(2,1). If it crashed, it's used!